lib/ansiblelint/cli.py

```diff
@@ -204,8 +204,29 @@ def get_config(arguments):
     options = parser.parse_args(arguments)
 
     config = load_config(options.config_file)
+    config = merge_config(config, options)
+
+    if config.listrules or config.listtags:
+        return config
+
+    # Exclude playbooks according to exclude path.
+    for playbook in list(config.playbook):
+        for excluded in config.exclude_paths:
+            try:
+                playbook.relative_to(excluded)
+            except ValueError:
+                pass
+            else:
+                config.playbook.remove(playbook)
+
+    # no args triggers auto-detection mode
+    if not config.playbook:
+        config.playbook = get_playbooks_and_roles(options=config)
+        if not config.playbook:
+            print_help(file=sys.stderr)
+            return 1
```

Comment:
It's weird to return int where every other place returns dict. Did you mean to raise an exception maybe?

Reply:
I just moved code from the `main()` function to here, and forgot to change that. But yes, I guess raising `SystemExit()` would be more explicit.

Reply:
Right, but also it'd be great to consolidate places where we terminate the program. So that they are not distributed across multiple modules.

Reply:
The only other places where the program exits I could find are:
- in the `__main__` module;
- in the `utils` module;

Those in in the `__main__` module, I believe could be removed (and those don't use the constants you defined for return codes).

In the `utils` module, I find the use of `SystemExit()` debatable. If a file cannot be found, cannot be parsed, _etc._ that should be reported as an error like any other, no reason to stop the linting of other files. I generally prefer linting tools to spit out as much issues as they can so I can fix them in batch, without the need to run the command again and again. Maybe that's just me. 

That would put all termination points in the `cli` module but for when the main function returns.

If we want to change that, can we do it in another PR ? Let's stay focus on the issue at hand which, I believe, prevents the next release of the tool. The above are only internal changes that don't have any impact on the users.